src/layouts/sidebar-sidecar/utils/product-url-adjusters.ts

```diff
@@ -1,124 +1,174 @@
-import { Pluggable } from 'unified'
-import remarkPluginAdjustLinkUrls from 'lib/remark-plugin-adjust-link-urls'
 import isAbsoluteUrl from 'lib/is-absolute-url'
+import { ProductData, ProductSlug } from 'types/products'
+// product data
+import boundaryData from 'data/boundary.json'
+import consulData from 'data/consul.json'
+import hcpData from 'data/hcp.json'
+import nomadData from 'data/nomad.json'
+import packerData from 'data/packer.json'
 import sentinelData from 'data/sentinel.json'
+import terraformData from 'data/terraform.json'
+import vagrantData from 'data/vagrant.json'
+import vaultData from 'data/vault.json'
+import waypointData from 'data/waypoint.json'
 
 /**
+ * Dictionary of product slugs,
+ * to (url) => adjustedUrl functions.
+ *
+ * Note that these URL adjustments are expected to be necessary for
+ * every product during initial migration to HashiCorp Developer.
+ *
+ * If we want to remove the need for these URL adjustmenets, then
+ * at some point in the future, we could apply these as remark transforms
+ * using "remarkPluginAdjustLinkUrls" to the original source content
+ * (across ALL versions!), and ask authors to use the adjusted URL format
+ * going forward. Asana task:
+ * https://app.asana.com/0/1202097197789424/1202569127502507/f
+ */
+const productUrlAdjusters: Record<ProductSlug, (url: string) => string> = {
+	boundary: (url) => rewriteDocsUrl(url, boundaryData as ProductData),
+	consul: (url) => {
+		const withApiDocsFix = rewriteApiToApiDocs(url)
+		const withPrefix = rewriteDocsUrl(withApiDocsFix, consulData as ProductData)
+		return withPrefix
+	},
+	hcp: (url) => rewriteDocsUrl(url, hcpData as ProductData),
+	nomad: (url) => rewriteDocsUrl(url, nomadData as ProductData),
+	packer: (url) => rewriteDocsUrl(url, packerData as ProductData),
+	sentinel: rewriteSentinelDocsUrls,
+	terraform: (url) => rewriteDocsUrl(url, terraformData as ProductData),
+	vagrant: (url) => rewriteDocsUrl(url, vagrantData as ProductData),
+	vault: (url) => {
+		const withApiDocsFix = rewriteApiToApiDocs(url)
+		const withPrefix = rewriteDocsUrl(withApiDocsFix, vaultData as ProductData)
```

Comment:
Vault and Consul already had `rewriteApiToApiDocs` applied. This was applied before rendering on the client. So, the ordering of URL adjustments is retained:
1. fix `/api` → `/api-docs`
2. prefix with product slug (previously done client-side)